### PR TITLE
DAOS-6885 Telemetry: Added DER_SHMEM_PERMS

### PR DESCRIPTION
**Pull request by jbrosenz**
Added DER_SHMEM_PERMS which is now used to differentiate
    between a shared memory permissions failure and an out of
    shared memory error.

    Added steps to the admin guide troubleshooting section that
    describes what the DER_SHMEM_PERMS error means and how to
    correct it.

Signed-off-by: Joel Rosenzweig <joel.b.rosenzweig@intel.com>

### CODE REVIEW

File: doc/admin/troubleshooting.md
Comment (kjacque) [inline doc/admin/troubleshooting.md:298]: Instead of "DAOS staff," would be better to say "DAOS maintainers"

File: doc/admin/troubleshooting.md
Comment (kjacque) [inline doc/admin/troubleshooting.md:250]: This would be a good time to consider adding subheadings to this section of the doc. "Incompatible Agent", "HLC Sync", "Shared Memory Errors"

**Review by kjacque — COMMENTED** [review]
This doc update is really helpful. I appreciate the detail! A couple minor suggestions, overall looks good though.

**Review by tanabarr — DISMISSED** [review]

File: doc/admin/troubleshooting.md
Comment (jbrosenz) [reply doc/admin/troubleshooting.md]: Done

File: doc/admin/troubleshooting.md
Comment (jbrosenz) [reply doc/admin/troubleshooting.md:252]: Done

**Review by tanabarr — APPROVED** [review]

**Review by kjacque — APPROVED** [review]

**Comment by jbrosenz** [comment]
I will add a new error code for a permissions error to distinguish between that and out of space.  I'll push a revised doc update at the same time.  Let's hold off on landing this doc update until then.

File: doc/admin/troubleshooting.md
Comment (kjacque) [inline doc/admin/troubleshooting.md:291]: Any need to leave in discussion of the other case, where the shared memory is full?

**Review by kjacque — DISMISSED** [review]

File: doc/admin/troubleshooting.md
Comment (jbrosenz) [reply doc/admin/troubleshooting.md:291]: My thought was that it was not necessary because it really ought to be caught by developers.  I had it in originally for completeness, but I think that now there is a specific error code for the permissions failure, it is cleaner this way.

Comment (kjacque) [reply doc/admin/troubleshooting.md:291]: Okay, works for me. Thanks.

File: doc/admin/troubleshooting.md
Comment (tanabarr) [inline doc/admin/troubleshooting.md:252]: I'm sure there are inconsistencies but I think we should try to keep to "I/O Engine" at least when adding new text.

File: src/gurt/telemetry.c
Comment (tanabarr) [inline src/gurt/telemetry.c:2371]: nit: this error message slightly inconsistent with the one above, here you just mention the function name that failed were above you elaborate, "unable to detach…" rather than "shmat…"

**Review by tanabarr — DISMISSED** [review]

File: doc/admin/troubleshooting.md
Comment (jbrosenz) [reply doc/admin/troubleshooting.md:252]: Yes, that's a good change.  I will fix that.

File: src/gurt/telemetry.c
Comment (jbrosenz) [reply src/gurt/telemetry.c]: I've made some updates there to elaborate and make them consistent.  See what you think.

**Review by kjacque — APPROVED** [review]

**Review by tanabarr — APPROVED** [review]

**Comment by jbrosenz** [comment]
Two test failures: are DAOS-6932.  
Scan CentOS 7 RPMs, CORCI-1055

**Pull request merged**
